indico/modules/events/papers/models/revisions.py

```diff
@@ -182,3 +182,21 @@ def get_reviewed_for_groups(self, user, include_reviewed=False):
         if self.paper.cfp.layout_reviewing_enabled and user in self.paper.cfp.layout_reviewers:
             reviewed_for.add(PaperReviewType.layout)
         return set(map(PaperTypeProxy, reviewed_for))
+
+    def has_user_reviewed(self, user, review_type=None):
+        from indico.modules.events.papers.models.reviews import PaperReviewType
+        if review_type:
+            if isinstance(review_type, basestring):
+                review_type = PaperReviewType[review_type]
+            return any(review.user == user and review.type == review_type for review in self.reviews)
+        else:
+            layout_review = next((review for review in self.reviews
+                                  if review.user == user and review.type == PaperReviewType.layout), None)
+            content_review = next((review for review in self.reviews
+                                   if review.user == user and review.type == PaperReviewType.content), None)
+            if user in self._contribution.paper_layout_reviewers and user in self._contribution.paper_content_reviewers:
+                return layout_review and content_review
+            elif user in self._contribution.paper_layout_reviewers:
+                return layout_review
+            elif user in self._contribution.paper_content_reviewers:
+                return content_review
```

Comment:
Why does a method called `has_*` return something that is not a boolean? I think all 3 returns should be wrapped in a bool() or have `is not None` added. And probably an `else: return False` (currently you return None in that case)